types/resource_selector.go

```diff
@@ -9,13 +9,26 @@ import (
 	"github.com/flanksource/commons/collections"
 	"github.com/flanksource/commons/hash"
 	"github.com/flanksource/commons/logger"
+	"github.com/lib/pq"
 	"gorm.io/gorm"
 	"gorm.io/gorm/clause"
 	"gorm.io/gorm/schema"
 	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/apimachinery/pkg/labels"
 )
 
+type ComponentConfigTraversalArgs struct {
+	ComponentID string         `yaml:"component_id,omitempty" json:"component_id,omitempty"`
+	Direction   string         `yaml:"direction,omitempty" json:"direction,omitempty"`
+	Types       pq.StringArray `yaml:"types,omitempty" json:"types,omitempty"`
```

Comment:
should use resourceSelector.types instead

Reply:
These types go directly into the postgres function, I'll see if the `Items` type work else I'll derive that from pq.StringArray